### PR TITLE
marsRover60: Renders nav bar appropriate nav bar elements for signed in and signed out user

### PR DESCRIPTION
**Pull request by HeatherJLewis**
**Description**
marsRover 60: Should remove login link from the homepage when the user is logged in

**Changes**
* refactors nav bar to use a div rather than a ul - this simplifies the JS to add elements when user is signed in
* renders the `login` and `register` link in the nav bar when the user is signed out
* renders the `account`, `rover`, `calculator` and `logout` link when the user is signed in

### CODE REVIEW

File: app/homepage/index.js [hunk 1]
```diff
@@ -73,13 +73,53 @@ const getUsername = () => {
...
+
+			const userAccountLink = document.createElement('a');
+			userAccountLink.href = '/user/account';
+			userAccountLink.textContent = 'Account';
+			roverMapList.appendChild(userAccountLink);
+
+			const logoutLink = document.createElement('a');
+			logoutLink.href = '/user/logout';
+			logoutLink.textContent = 'Logout';
+			roverMapList.appendChild(logoutLink);
```
Comment (Kenneth-Siu) [inline app/homepage/index.js]: Interesting! I like the navbar that changes links based on whether or not the user's logged in!

My first thought was that we should see if we can't DRY this bit out.

My second thought is to do with separation between HTML and JS: let's put our structural elements in the HTML — i.e. start with all the links in the HTML — and then apply customised CSS classes to them that depend on the user's logged-in status — i.e. update their CSS classes in the JS.

File: app/homepage/index.js [hunk 2]
```diff
@@ -73,13 +73,53 @@ const getUsername = () => {
...
+	const navbar = document.getElementById('navbar');
+	const offsetPositionOfNavBar = navbar.offsetTop;
+
+	const toggleStickyClass = () => {
+		if (window.pageYOffset >= offsetPositionOfNavBar) {
+			navbar.classList.add('sticky');
+		} else {
+			navbar.classList.remove('sticky');
+		}
+	};
```
Comment (Kenneth-Siu) [inline app/homepage/index.js]: My suspicion of what would happen is that when you start scrolling, the rest of the page will jump up a number of pixels equal to the height of the navbar. Is that true? Have you already resolved this problem? Would I be able to take a look?

File: app/login/index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+			navbar.classList.add('sticky');
+		} else {
+			navbar.classList.remove('sticky');
+		}
+	};
+
+	window.onscroll = () => toggleStickyClass();
+};
+
+stickyNavBar();
```
Comment (Kenneth-Siu) [inline app/login/index.js]: I think I might prefer to just have a navbar script that both the homepage and the login page load. (Of course this type of problem doesn't exist with a compilation pipeline and/or react, but such is vanilla JS!)

File: app/homepage/index.html [hunk 4]
```diff
@@ -7,6 +7,11 @@
         <link rel="stylesheet" href="./styles.css">
     </head>
     <body>
+        <div id="navbar">
+            <a class="active" href="/">Home</a>
+            <a id="login-link" href="/login">Login</a>
+            <a id="register-link" href="/register">Register</a>
+        </div>
```
Comment (Kenneth-Siu) [inline app/homepage/index.html]: You've already said to me about the `<nav>` element :)

**Review by Kenneth-Siu — COMMENTED** [review]
Good to see thinking about semantic HTML!

A few thoughts!

File: app/homepage/index.html (hunk 4 above)
Comment (HeatherJLewis) [reply app/homepage/index.html]: Done :)

File: app/homepage/index.js (hunk 2 above)
Comment (HeatherJLewis) [reply app/homepage/index.js]: Fixed. Thank you.

File: app/homepage/index.js (hunk 1 above)
Comment (HeatherJLewis) [reply app/homepage/index.js]: Discussed and resolved.

**Review by Kenneth-Siu — COMMENTED** [review]
I think the only thing I'd change is to do away with the css class `sticky` and just put that as part of the navbar's css

Otherwise, it looks excellent — if you make the final css change (remember to update both html files) I'm happy for you to just merge it in

**Pull request merged**